aten/src/ATen/native/vulkan/VulkanAten.cpp

```diff
@@ -163,6 +164,95 @@ at::Tensor vulkan_adaptive_avg_pool2d(
   return output;
 }
 
+at::Tensor vulkan_max_pool2d(
+    const at::Tensor& self,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    bool ceil_mode) {
+  TORCH_CHECK(
+      kernel_size.size() == 1 || kernel_size.size() == 2,
+      "Vulkan max_pool2d: kernel_size must either be a single int, or a tuple of two ints")
+  const int kH = safe_downcast<int, int64_t>(kernel_size[0]);
```

Comment:
This usage is technically correct but [template argument deduction](https://en.cppreference.com/w/cpp/language/template_argument_deduction) can deduce the second template parameter (i.e. source type) passed as a function argument, so this will work as well:

```
const int kH = safe_downcast<int>(kernel_size[0]); 
```